sui/its.js

```diff
@@ -19,48 +19,43 @@ function parseTrustedChains(config, trustedChain) {
 }
 
 async function setupTrustedAddress(keypair, client, config, contracts, args, options) {
-    const [trustedChain, trustedAddress] = args;
+    const [trustedChain] = args;
 
-    const { ITS: itsConfig } = contracts;
+    const { InterchainTokenService: itsConfig } = contracts;
 
-    const { OwnerCap, ITS } = itsConfig.objects;
+    const { OwnerCap, InterchainTokenService } = itsConfig.objects;
 
     const txBuilder = new TxBuilder(client);
 
     const trustedChains = parseTrustedChains(config, trustedChain);
 
-    const trustedAddressesObject = await txBuilder.moveCall({
-        target: `${itsConfig.address}::trusted_addresses::new`,
-        arguments: [trustedChains, trustedChains.map(() => trustedAddress)],
-    });
-
     await txBuilder.moveCall({
-        target: `${itsConfig.address}::its::set_trusted_addresses`,
-        arguments: [ITS, OwnerCap, trustedAddressesObject],
+        target: `${itsConfig.address}::interchain_token_service::add_trusted_chains`,
+        arguments: [InterchainTokenService, OwnerCap, trustedChains],
     });
 
     if (options.offline) {
         const tx = txBuilder.tx;
         const sender = options.sender || keypair.toSuiAddress();
         tx.setSender(sender);
-        await saveGeneratedTx(tx, `Set trusted address for ${trustedChain} to ${trustedAddress}`, client, options);
+        await saveGeneratedTx(tx, `Added trusted chain ${trustedChain}`, client, options);
     } else {
         await broadcastFromTxBuilder(txBuilder, keypair, 'Setup Trusted Address');
     }
 
-    // Update ITS config
+    // Update InterchainTokenService config
     for (const trustedChain of trustedChains) {
-        // Add trusted address to ITS config
-        if (!contracts.ITS.trustedAddresses) contracts.ITS.trustedAddresses = {};
+        // Add trusted address to InterchainTokenService config
+        if (!contracts.InterchainTokenService.trustedAddresses) contracts.InterchainTokenService.trustedAddresses = [];
 
-        contracts.ITS.trustedAddresses[trustedChain] = trustedAddress;
+        contracts.InterchainTokenService.trustedAddresses.push(trustedChain);
```

Comment:
we shouldn't be storing the list of trustedAddress/chains separately in the config I think (we don't do it for EVM ITS). Maybe add a query in another PR if it doesn't exist to get the full list of trusted chains from Sui ITS?

Reply:
Removed altogether, we do not have getters for anything not used by another one of our contracts in Sui.

Reply:
Can we read it off the object itself though in this case?